src/stepFunctions/activation.ts

```diff
@@ -110,22 +110,23 @@ function initializeCodeLens(context: vscode.ExtensionContext) {
                 command: 'aws.previewStateMachine',
                 title: localize('AWS.stepFunctions.render', 'Render graph'),
             }
-
-            const publishCommand: vscode.Command = {
-                command: 'aws.stepfunctions.publishStateMachine',
-                title: localize('AWS.stepFunctions.publish', 'Publish to Step Functions'),
-            }
-
             const renderCodeLens = new vscode.CodeLens(topOfDocument, renderCommand)
-            const publishCodeLens = new vscode.CodeLens(topOfDocument, publishCommand)
 
-            return [publishCodeLens, renderCodeLens]
+            if (document.languageId === 'asl') {
+                const publishCommand: vscode.Command = {
+                    command: 'aws.stepfunctions.publishStateMachine',
+                    title: localize('AWS.stepFunctions.publish', 'Publish to Step Functions'),
+                }
+                const publishCodeLens = new vscode.CodeLens(topOfDocument, publishCommand)
+
+                return [publishCodeLens, renderCodeLens]
+            } else {
+                return [renderCodeLens]
```

Comment:
Just curious, is there a reason why `asl-yaml` isn't valid for publish? We're seeing a decent amount of publish usage in VS Code and I wonder if users will expect this functionality for YAML.

Reply:
We could technically trasform the yaml to json and publish it. We may want to add yaml support directly in our api later though in which case it could be a bit confusing. Let me consult with our PM. If we want to add it I'll make another PR.